core/kazoo_convert/src/kz_fax_converter.erl

```diff
@@ -0,0 +1,401 @@
+    ];
+eval_format(CT, ?TIFF_MIME) when ?OPENOFFICE_COMPATIBLE(CT) ->
+    [fun openoffice_to_pdf/2
+    ,fun pdf_to_tiff/2
+    ];
+eval_format(<<?OPENOFFICE_MIME_PREFIX, _/binary>>, ?TIFF_MIME) ->
+    [fun openoffice_to_pdf/2
+    ,fun pdf_to_tiff/2
+    ];
+eval_format(<<?OPENXML_MIME_PREFIX, _/binary>>, ?PDF_MIME) ->
+    [fun openoffice_to_pdf/2
+    ];
+eval_format(<<?OPENOFFICE_MIME_PREFIX, _/binary>>, ?PDF_MIME) ->
+    [fun openoffice_to_pdf/2
+    ];
+eval_format(CT, ?PDF_MIME) when ?OPENOFFICE_COMPATIBLE(CT) ->
+    [fun openoffice_to_pdf/2
+    ];
+eval_format(FromFormat, ToFormat) ->
+    {'error', <<"invalid conversion requested: ", FromFormat/binary, " to: ", ToFormat/binary>>}.
+
+-spec run_convert({'error', kz_term:ne_binary()} | fax_convert_funs()
+                 ,kz_term:ne_binary()
+                 ,kz_term:ne_binary()
+                 ,map()) -> gen_kz_converter:converted().
+run_convert({'error', _}=Error, _ToFormat, _FilePath, _Options) ->
+    Error;
+run_convert([Operation|Operations], ToFormat, FilePath, Options) ->
+    case Operation(FilePath, Options) of
+        {'ok', OutputPath} ->
+            maybe_delete_previous_file(FilePath, OutputPath),
+            run_convert(Operations, ToFormat, OutputPath, Options);
+        Error -> Error
+    end;
+run_convert([], ToFormat, FilePath, Options) ->
+    case validate_output(ToFormat, FilePath, Options) of
+        {'ok', _} ->
+            format_response(ToFormat, FilePath, Options);
+        Error -> Error
+    end.
+
+-spec image_to_tiff(kz_term:ne_binary(), map()) -> fax_converted().
+image_to_tiff(FromPath, Options) ->
+    convert_file(?CONVERT_IMAGE_COMMAND, FromPath, <<".tiff">>, Options).
+
+-spec tiff_to_pdf(kz_term:ne_binary(), map()) -> fax_converted().
+tiff_to_pdf(FromPath, Options) ->
+    convert_file(?CONVERT_TIFF_COMMAND, FromPath, <<".pdf">>, Options).
+
+-spec pdf_to_tiff(kz_term:ne_binary(), map()) -> fax_converted().
+pdf_to_tiff(FromPath, Options) ->
+    convert_file(?CONVERT_PDF_COMMAND, FromPath, <<".tiff">>, Options).
+
+-spec openoffice_to_pdf(kz_term:ne_binary(), map()) -> fax_converted().
+openoffice_to_pdf(FromPath, Options) ->
+    case ?ENABLE_OPENOFFICE of
+        'true' ->
+            case ?SERIALIZE_OPENOFFICE of
+                'true' -> kz_openoffice_server:add(FromPath, Options);
+                'false' -> do_openoffice_to_pdf(FromPath, Options)
+            end;
+        'false' ->
+            {'error', <<"openoffice compatible conversion is unsupported">>}
+    end.
+
+-spec do_openoffice_to_pdf(kz_term:ne_binary(), map()) -> fax_converted().
+do_openoffice_to_pdf(FromPath, Options) ->
+    convert_file(?CONVERT_OPENOFFICE_COMMAND, FromPath, <<".pdf">>, Options).
+
+-spec convert_file(kz_term:ne_binary(), kz_term:ne_binary(), kz_term:ne_binary(), map()) -> fax_converted().
+convert_file(Command, FromPath, Ext, #{<<"job_id">> := JobId, <<"tmp_dir">> := TmpDir}) ->
+    ToPath = filename:join(TmpDir, [JobId, Ext]),
```

Comment:
Why not just do `filename:join([TmpDir, JobId, Ext]),`?

Reply:
because:
```erlang
1> filename:join("/tmp", ["sdsdw34", ".pdf"]).
"/tmp/sdsdw34.pdf"
2> filename:join(["/tmp", "sdsdw34", ".pdf"]). 
"/tmp/sdsdw34/.pdf"
```

Reply:
This is why:
2> filename:join([<<"path">>, <<"file">>, <<".ext">>]).
<<"path/file/.ext">>
3> filename:join(<<"path">>, [<<"file">>, <<".ext">>]).
<<"path/file.ext">>

Reply:
Ah, didn't realize they were not all path segments. Thanks!

Reply:
That said, I don't think Dialyzer likes it. Maybe append ext to jobid before filename:join?

Reply:
Something like this pleases the Dialyzer:

```
-    ToPath = filename:join(TmpDir, [JobId, Ext]),
-    BatchPath = filename:join(TmpDir, [filename:rootname(filename:basename(FromPath)), Ext]),
+    ToPath = filename:join([TmpDir, <<JobId/binary, Ext/binary>>]),
+    BatchPath = filename:join([TmpDir, <<(filename:rootname(filename:basename(FromPath)))/binary, Ext/binary>>]),
```